src/WebJobs.Script/Description/FunctionGroupListenerDecorator.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    internal class FunctionGroupListenerDecorator : IListenerDecorator
+    {
+        private static readonly NoOpListener _noOpListener = new();
+        private readonly IFunctionMetadataManager _metadata;
+        private readonly IEnvironment _environment; // TODO: replace options pattern
+        private readonly ILogger _logger;
+
+        public FunctionGroupListenerDecorator(
+            IFunctionMetadataManager metadata,
+            IEnvironment environment,
+            ILogger<FunctionGroupListenerDecorator> logger)
+        {
+            _metadata = metadata ?? throw new ArgumentNullException(nameof(metadata));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public IListener Decorate(ListenerDecoratorContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (!_environment.TryGetFunctionsTargetGroup(out string targetGroup))
+            {
+                // If no group configured, short-circuit.
+                return context.Listener;
+            }
+
+            _logger.LogInformation("Function group target is {targetGroup}", targetGroup);
+            string functionName = context.FunctionDefinition.Descriptor.ShortName;
+            if (!_metadata.TryGetFunctionMetadata(functionName, out FunctionMetadata functionMetadata))
+            {
+                _logger.LogWarning("Unable to find function metadata for function {functionName}", functionName);
+                return context.Listener;
+            }
+
+            string group = functionMetadata.GetFunctionGroup() ?? functionName;
+            if (string.Equals(targetGroup, group, StringComparison.OrdinalIgnoreCase))
+            {
+                _logger.LogDebug("Enabling function {functionName}", functionName);
+                return context.Listener;
+            }
+
+            // A target function group is configured and this function is not part of it.
+            // By giving a no-op listener, we will prevent it from triggering without 'disabling' it.
+            _logger.LogDebug("Function {functionName} is not part of group {functionGroup}. Listener will not be enabled.", functionName, targetGroup);
+            context.Listener?.Dispose(); // this will not be used, lets dispose it now.
```

Comment:
Need to think about this - the platform [here](https://github.com/Azure/azure-webjobs-sdk/blob/1e731b26b33e4cb302e5a30447f2655e6e291b0f/src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs#L66) is actually holding onto a reference to this listener for scale monitoring purposes. I suspect that if you test this code and call the host scale status APIs you'll get a null ref

Reply:
Ah I see. Do we even want that code to run in this scenario?

Reply:
Updated so that we don't dispose until later.

Reply:
With the dispose change the scale endpoint still works (even when the listener was never started). However, will need to discuss if this endpoint is even used in per-func scaling. It does not appear to be designed for it, as there is only a single current worker count provided: _which group_ does that worker count represent?

Reply:
My main point is that the host believes it owns the listener instance and will dispose it. So I don't think you should dispose it.

Reply:
When looking at other decorator listeners, they hold a reference of the decorated instance and dispose it on their own disposal. I switched to that pattern to be in line with them.